c7n/resources/rdscluster.py

```diff
@@ -748,48 +751,132 @@ class PendingMaintenance(Filter):
                       - system-update
     """
 
-    annotation_key = 'c7n:PendingMaintenance'
-    schema = type_schema('pending-maintenance')
-    permissions = ('rds:DescribePendingMaintenanceActions',)
+    annotation_key = "c7n:PendingMaintenance"
+    schema = type_schema("pending-maintenance")
+    permissions = ("rds:DescribePendingMaintenanceActions",)
 
     def process(self, resources, event=None):
-        client = local_session(self.manager.session_factory).client('rds')
+        client = local_session(self.manager.session_factory).client("rds")
 
         results = []
         resource_maintenances = {}
-        paginator = client.get_paginator('describe_pending_maintenance_actions')
+        paginator = client.get_paginator("describe_pending_maintenance_actions")
         for page in paginator.paginate():
-            for action in page['PendingMaintenanceActions']:
-                resource_maintenances.setdefault(action['ResourceIdentifier'], []).append(action)
+            for action in page["PendingMaintenanceActions"]:
+                resource_maintenances.setdefault(action["ResourceIdentifier"], []).append(action)
 
         for r in resources:
-            pending_maintenances = resource_maintenances.get(r['DBClusterArn'], [])
+            pending_maintenances = resource_maintenances.get(r["DBClusterArn"], [])
             if len(pending_maintenances) > 0:
                 r[self.annotation_key] = pending_maintenances
                 results.append(r)
 
         return results
 
 
-class DescribeDbShardGroup(DescribeSource):
-    def augment(self, resources):
-        for r in resources:
-            r['Tags'] = r.pop('TagList', ())
-        return resources
-
+@RDSClusterSnapshot.action_registry.register("region-copy")
+class RegionCopyClusterSnapshot(BaseAction):
```

Comment:
I would suggest starting from a clean branch and just copying this action and the associated tests and data files.